src/Microsoft.AspNet.OData.Shared/EnableQueryAttribute.cs

```diff
@@ -751,8 +751,8 @@ internal static void ValidateSelectExpandOnly(ODataQueryOptions queryOptions)
                 throw Error.InvalidOperation(SRResources.QueryGetModelMustNotReturnNull);
             }
 
-            IEdmEntityType baseEntityType = model.GetEdmType(elementClrType) as IEdmEntityType;
-            IEdmStructuredType structuredType = model.GetEdmType(elementClrType) as IEdmStructuredType;
+            IEdmEntityType baseEntityType     = model.GetTypeMappingCache().GetEdmType(elementClrType, model)?.Definition as IEdmEntityType;
```

Comment:
>model.GetTypeMappingCache().GetEdmType(elementClrType, model) [](start = 48, length = 61)

can we save the middle value, it seems here duplicate the codes twice?

Reply:
Done!